azext_edge/edge/providers/check/base/pod.py

```diff
@@ -96,65 +93,159 @@ def _decorate_pod_condition(condition: bool) -> Tuple[str, str]:
         pod_dict = pod.to_dict()
         pod_name = pod_dict["metadata"]["name"]
         pod_phase = pod_dict.get("status", {}).get("phase")
-        pod_conditions = pod_dict.get("status", {}).get("conditions", [])
+        pod_conditions: list = pod_dict.get("status", {}).get("conditions", [])
         pod_phase_deco, status = decorate_pod_phase(pod_phase)
 
+        pod_eval_value = {}
+        pod_eval_status = status
+        pod_eval_value["status.phase"] = pod_phase
+
+        conditions_readiness = True
+        conditions_display_list: List[Tuple[str, str]] = []
+        unknown_conditions_display_list: List[Tuple[str, str]] = []
+
+        # When pod in obnormal state, sometimes the conditions are not available
+        if pod_conditions:
+            for condition in pod_conditions:
+                type = condition["type"]
```

Comment:
is "type" guaranteed to be in `condition`? The code on line 96 indicates conditions can be an empty array.

Reply:
the conditions can be empty, but when there is condition in pod_conditions(did a check in 108), the type will be there